app/controllers/kudos_controller.rb

```diff
@@ -24,6 +28,8 @@ def create
     kudo = Kudo.new(kudo_params)
     kudo.giver = current_employee
     if kudo.save
```

Comment:
you should check here as well if number_of_available_kudos > 1, because someone can:
- open multiple forms for submitting kudos (which will pass validation)
- submit them - now controller, won't check again, if the number has not decreased in the meantime

Reply:
I added a check condition in the create method